### PR TITLE
Hook up payment information to api

### PR DESCRIPTION
**Pull request by cvalarida**
### Success
image

@goldenmeanie I went ahead and made as many circles as there are digits we're masking. The mock had only 6 for each and showed the last 4 digits of both numbers. Should I switch to using 6 circles, or keep this?

Also, we could have a little button to unmask the number if we wanted. Is that a thing that would be useful?

### Error
image

@peggygannon What should this error say? The context for this is the following:
1) The user starts a form
2) Once they get to this page (a few screens in), we query for their most up-to-date payment information
3) The call failed for some reason
    - This could be a number of reasons, but they all mean the same thing to the user; they can't see their most up-to-date payment information
    - This does not stop them from continuing with the form
    - Once they submit, we'll make this same call again to submit their form with the payment information they have on file; if it fails at _that_ point, then we'll have to keep trying until it goes through
      - That might be more information than you or the user need to know, but in case it's helpful, there it is

### CODE REVIEW

**Comment by cvalarida** [comment]
To test this, you'll have to be connected to EVSS' VPN and have vets-api running with a logged-in user.

**Review by cehsu — APPROVED** [review]
LGTM, pending @peggygannon's input on the error message. Should any tests be updated/created?

File: src/applications/disability-benefits/526EZ/pages/paymentInfo.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    accountNumber,
+    financialInstitutionRoutingNumber: routingNumber,
+    financialInstitutionName: bankName,
+  } = response;
+  let accountNumberString;
+  let routingNumberString;
+  let bankNameString;
+  const mask = (string, unmaskedLength) => {
+    const maskedString = srSubstitute(`${'●'.repeat(string.length - unmaskedLength)}`, 'ending with');
+    return <span>{maskedString}{string.slice(unmaskedLength * -1)}</span>;
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/pages/paymentInfo.jsx]: Oh hey. For some reason, I thought that'd give me a syntax error so I didn't even try. Turns out it works.

Good call!

**Comment by cvalarida** [comment]
Yeah, there are no tests for this page; I'll write some. :+1:

**Comment by peggygannon** [comment]
@cvalarida Here are my edit to the bank account payment screen: 

1. Please update intro sentence to: 
--This is the bank account information we have on file for you. We pay your disability and pension benefits to this account. 
--Change Routing number TO: Bank routing number

2. I have a few followup questions for the error message: 
--In the message on this page, should we ask the user to do anything (i.e, update their payment info). Or are we just letting them know we can't find the payment information that's on the screen? 

--Do I need to craft another (more detailed message) that will appear later when they try to submit their form?

**Comment by cvalarida** [comment]
@peggygannon Good question. I'm not sure. Probably what we'll do is get a list of all the errors that the api will bubble back up if the submit doesn't go through. When we get that list, we'll notify the user in some way. At _that_ point, we'll need copy to be golden, but I think we need to start with getting a list of errors first.

@kreek Does that list of errors exist somewhere, or should we write a ticket to make it?

**Comment by peggygannon** [comment]
@cvalarida OK, thanks. Sounds like we need more information before we can craft a final message. In the meantime, can we drop in this error message and then I'll update the message as we get more info? 

**We can't access your information**
We're sorry. We can't access your payment information right now.  You can continue to fill out the form and we'll try again later. 


Also, I want to be sure you also saw my updates to the payment intro: 
Please update intro sentence to:
--This is the bank account information we have on file for you. We pay your disability and pension benefits to this account.
--Change Routing number TO: Bank routing number

**Comment by cvalarida** [comment]
Oh, yes, I did see those changes. Just made them. To answer your second question (which I completely ignored earlier--sorry!), we're just letting them know we can't find it at the moment. :+1:

**Comment by cvalarida** [comment]
@peggygannon Here's the updated screenshot.
image

**Review by peggygannon — APPROVED** [review]
@cvalarida LGTM! thanks!

**Pull request merged**